server/client.go

```diff
@@ -2132,17 +2130,18 @@ func (c *client) processInboundClientMsg(msg []byte) {
 			atomic.LoadInt64(&c.srv.gateway.totalQSubs) > 0 {
 			qnames = &queues
 		}
-		replySub = c.processMsgResults(c.acc, r, msg, c.pa.subject, c.pa.reply, qnames)
+		c.processMsgResults(c.acc, r, msg, c.pa.subject, c.pa.reply, qnames)
 	}
 
 	// Now deal with gateways
 	if c.srv.gateway.enabled {
 		// TODO(ik): Need to revisit all that.
-		// If replySub is not nil it means that this is
-		// a reply sent on the _R_ subject and associated with
-		// an outbound connection. Send direct here.
-		if replySub != nil {
-			c.sendReplyMsgDirectToGateway(c.acc, replySub, msg)
+		// If we have a single match and the sub's client connection
+		// is a gateway, we know it is a message sent to a "_R_." subject
+		// (since this is the only time we add a sub with a GW client
+		// into an account sublist). In that case, do a direct send.
+		if len(r.psubs) == 1 && r.psubs[0].client.gw != nil {
+			c.sendReplyMsgDirectToGateway(c.acc, r.psubs[0], msg)
```

Comment:
This only one right? Will this break anything?

Reply:
See previous comment. For now this is correct behavior based on agreed contract.